detector/aws_security_group_duplicate_name_test.go

```diff
@@ -0,0 +1,227 @@
+					IsDefault: aws.Bool(true),
+				},
+			},
+			Issues: []*issue.Issue{},
+		},
+		{
+			Name: "security group name is duplicate in another vpc when omitted vpc_id",
+			Src: `
+resource "aws_security_group" "test" {
+    name   = "default"
+}`,
+			SecurityGroups: []*ec2.SecurityGroup{
+				&ec2.SecurityGroup{
+					GroupName: aws.String("default"),
+					VpcId:     aws.String("vpc-1234abcd"),
+				},
+				&ec2.SecurityGroup{
+					GroupName: aws.String("custom"),
+					VpcId:     aws.String("vpc-1234abcd"),
+				},
+			},
+			Vpcs: []*ec2.Vpc{
+				&ec2.Vpc{
+					VpcId:     aws.String("vpc-1234abcd"),
+					IsDefault: aws.Bool(true),
+				},
+				&ec2.Vpc{
+					VpcId:     aws.String("vpc-abcd1234"),
+					IsDefault: aws.Bool(false),
+				},
+			},
+			Issues: []*issue.Issue{
+				&issue.Issue{
+					Type:    "ERROR",
+					Message: "\"default\" is duplicate name. It must be unique.",
+					Line:    3,
+					File:    "test.tf",
+				},
+			},
+		},
+		{
+			Name: "security group name is duplicate, but exists in state",
+			Src: `
+resource "aws_security_group" "test" {
+    name   = "default"
+    vpc_id = "vpc-1234abcd"
+}`,
+			State: `
+{
+    "modules": [
+        {
+            "resources": {
+                "aws_security_group.test": {
+                    "type": "aws_security_group",
+                    "depends_on": [],
+                    "primary": {
+                        "id": "sg-1234abcd",
+                        "attributes": {
+                            "id": "sg-1234abcd",
+                            "name": "default",
+                            "owner_id": "123456789",
+                            "vpc_id": "vpc-1234abcd"
+                        }
+                    }
+                }
+            }
+        }
+    ]
+}
+`,
+			SecurityGroups: []*ec2.SecurityGroup{
+				&ec2.SecurityGroup{
+					GroupName: aws.String("default"),
+					VpcId:     aws.String("vpc-1234abcd"),
+				},
+				&ec2.SecurityGroup{
+					GroupName: aws.String("custom"),
+					VpcId:     aws.String("vpc-1234abcd"),
+				},
+			},
+			Issues: []*issue.Issue{},
+		},
+	}
+
+	for _, tc := range cases {
+		c := config.Init()
+		c.DeepCheck = true
+
+		awsClient := c.NewAwsClient()
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+		ec2mock := mock.NewMockEC2API(ctrl)
+		ec2mock.EXPECT().DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{}).Return(&ec2.DescribeSecurityGroupsOutput{
+			SecurityGroups: tc.SecurityGroups,
+		}, nil)
+		if tc.Vpcs != nil {
+			ec2mock.EXPECT().DescribeVpcs(&ec2.DescribeVpcsInput{}).Return(&ec2.DescribeVpcsOutput{
+				Vpcs: tc.Vpcs,
+			}, nil)
+		}
+		awsClient.Ec2 = ec2mock
+
+		var issues = []*issue.Issue{}
+		TestDetectByCreatorName(
+			"CreateAwsSecurityGroupDuplicateDetector",
+			tc.Src,
+			tc.State,
+			c,
+			awsClient,
+			&issues,
+		)
+
+		if !reflect.DeepEqual(issues, tc.Issues) {
+			t.Fatalf("Bad: %s\nExpected: %s\n\ntestcase: %s", issues, tc.Issues, tc.Name)
```

Comment:
__[gometalinter]__
* [vet] arg issues for printf verb %s of wrong type: []*github.com/wata727/tflint/issue.Issue

<!-- Comment by [SideCI](https://sideci.com) -->